python/ray/rllib/env/external_multi_agent_env.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from six.moves import queue
+import threading
+import uuid
+
+from ray.rllib.utils.annotations import PublicAPI
+from ray.rllib.env.external_env import ExternalEnv
+
+
+@PublicAPI
+class ExternalMultiAgentEnv(ExternalEnv):
+    """This is the multi-agent version of ExternalEnv."""
+
+    @PublicAPI
+    def __init__(self, action_space, observation_space, max_concurrent=100):
+        """Initialize a multi-agent external env.
+
+        ExternalMultiAgentEnv subclasses must call this during their __init__.
+
+        Arguments:
+            action_space (gym.Space): Action space of the env.
+            observation_space (gym.Space): Observation space of the env.
+            max_concurrent (int): Max number of active episodes to allow at
+                once. Exceeding this limit raises an error.
+        """
+        ExternalEnv.__init__(
+            self, action_space, observation_space, max_concurrent)
+
+        # we require to know all agents' spaces
+        if isinstance(self.action_space, dict) or isinstance(self.observation_space, dict):
+            assert(self.action_space.keys() == self.observation_space.keys())
```

Comment:
Instead of an assert, can you raise e.g., ValueError("Agent ids disagree for action space and obs space dict: {} {}".format(self.action_space.keys(), self.observation_space.keys())